crates/sui-framework/packages/dwallet-system/sources/dwallet_transfer.move

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) dWallet Labs, Ltd.
+// SPDX-License-Identifier: BSD-3-Clause-Clear
+
+#[allow(unused_use)]
+module dwallet_system::dwallet_transfer {
+    use dwallet::object::{Self, ID, UID};
+    use dwallet::token::recipient;
+    use dwallet::transfer;
+    use dwallet::tx_context;
+    use dwallet::tx_context::TxContext;
+    use dwallet_system::dwallet_2pc_mpc_ecdsa_k1::{DWallet, output};
+
+    const DWALLET_TRANSFER_ERROR: u64 = 0x1;
+
+    struct DwalletTransfer has key{
+        id: UID,
+        dwallet_id: ID,
+        encrypted_secret_share: vector<u8>,
+    }
+
+    struct PublicKey has key {
+        id: UID,
+        public_key: vector<u8>,
+        key_owner_address: address,
+    }
+
+    public fun store_public_key(key: vector<u8>, ctx: &mut TxContext): ID {
+        let pk = PublicKey {
+            id: object::new(ctx),
+            public_key: key,
+            key_owner_address: tx_context::sender(ctx),
+        };
+        let pk_id = object::id(&pk);
+        transfer::freeze_object(pk);
```

Comment:
A few consequences of this design:
1. There could be more than one encryption key per-address. There isn't really any need for that.
2. In case the decryption key has been lost, there is no way to "update" or delete this object as it is frozen. This means that in edge cases like these a dWallet could be transferred but the user could not access its secret share.
3. How do encrypters choose which encryption key to encrypt to?
4. Not every address is guranteed to hold an encryption key (I'm not sure actually that we can solve this, at least not without really changing SUI, so this might be ok)

Reply:
Alt #1: make this a shared object

Reply:
Alt #2: create a "registry" of encryption keys, which holds the current encryption key for every address. This could be a shared object, which is fine IMO since dWallet transfers don't seem to be a super time-critical task.

Reply:
@omersadika